src/main/java/com/ning/http/client/multipart/PartBase.java

```diff
@@ -89,11 +95,11 @@ protected void visitStart(PartVisitor visitor, byte[] boundary) throws IOExcepti
     protected void visitDispositionHeader(PartVisitor visitor) throws IOException {
         visitor.withBytes(CRLF_BYTES);
         visitor.withBytes(CONTENT_DISPOSITION_BYTES);
-        visitor.withBytes(getDispositionType() != null ? getDispositionType().getBytes(US_ASCII) : FORM_DATA_DISPOSITION_TYPE_BYTES);
+        visitor.withBytes(getDispositionType() != null ? getDispositionType().getBytes(getHeadersCharset()) : FORM_DATA_DISPOSITION_TYPE_BYTES);
```

Comment:
It seems a lot of overhead to be asking this all the time when the property is defined statically.